src/sdam/monitor.ts

```diff
@@ -486,42 +530,48 @@ export class RTTPinger {
   }
 }
 
-function measureRoundTripTime(rttPinger: RTTPinger, options: RTTPingerOptions) {
-  const start = now();
-  options.cancellationToken = rttPinger[kCancellationToken];
-  const heartbeatFrequencyMS = options.heartbeatFrequencyMS;
-
+function measureAndReschedule(
+  rttPinger: RTTPinger,
+  options: RTTPingerOptions,
+  start?: number,
+  conn?: Connection
+) {
```

Comment:
Refactored to use fewer closures and clarify control flow, but willing to retain the previous structure

Reply:
Fine by me, while we're at it, why not make this a class method, same with `measureRoundTripTime`